include/ocvsmd/sdk/node_pub_sub.hpp

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+//
+
+#ifndef OCVSMD_SDK_NODE_PUB_SUB_HPP_INCLUDED
+#define OCVSMD_SDK_NODE_PUB_SUB_HPP_INCLUDED
+
+#include "execution.hpp"
+
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <memory>
+
+namespace ocvsmd
+{
+namespace sdk
+{
+
+class RawSubscriber
+{
+public:
+    using Ptr = std::shared_ptr<RawSubscriber>;
+
+    virtual ~RawSubscriber() = default;
+
+    // No copy/move semantics.
+    RawSubscriber(RawSubscriber&&)                 = delete;
+    RawSubscriber(const RawSubscriber&)            = delete;
+    RawSubscriber& operator=(RawSubscriber&&)      = delete;
+    RawSubscriber& operator=(const RawSubscriber&) = delete;
+
+    struct Receive final
+    {
+        struct Success
+        {
+            std::size_t                   size;
+            std::unique_ptr<cetl::byte[]> data;  // NOLINT(*-avoid-c-arrays)
```

Comment:
Hmm, can this use PolymorphicDeleter so we have the opportunity to make this zero-copy or use other memory semantics other than global heap?

Reply:
I'm currently focused to deliver functionality. I believe it could be refactored later if there will be need of this. At my current stage it will unnecessarily complicate/delay things. So, if you don't mind, I'll just add comment here that we might reconsider the data memory storage/management later.